### DISCUSSION TITLE
Fullscreen with different resolutions doesn't work

### DISCUSSION THREAD

**Discussion by dozent** [KhronosGroup/MoltenVK#1304] [Q&A] [answered]
I want to implement different resolutions for a game in fullscreen mode. But vkGetPhysicalDeviceSurfaceCapabilitiesKHR return to me only one resolution. After searching I see that Vulkan takes a size from frame (NSWindow, NSView, CALayer). But If I set the size to frame the game rendering not at all screen, only on part of it. This is not what I want.
I can change the resolution by the method
`- (void)mtkView:(nonnull MTKView *)view drawableSizeWillChange:(CGSize)size;`
in Metal sample. But It does not work with Vulkan (MoltenVK).
I use Cocoa + MoltenVK.
So how implement different resolutions?

**Comment by cdavis5e** [comment, marked as answer]
If you want to do different resolutions, you can use Quartz Display Services to change the resolution. Then you can create a fullscreen `NSWindow` and `NSView` for that resolution, or resize one you already have to that resolution. `MTKView` does much of the work for you, but I don't see how it handles fullscreen resolution changes. Can you post your modified sample so I can see what you're talking about?

One of these days, I should implement `VK_KHR_display` and `VK_KHR_display_swapchain` so that `vkGetPhysicalDeviceSurfaceCapabilitiesKHR()` can report multiple resolutions for a fullscreen surface.

**Reply by dozent** [reply]
Thanks! Using Quartz Display Services helped me.
